frontend/hooks/useLensToken.js

```diff
@@ -0,0 +1,99 @@
+import { useEffect, useCallback } from "react";
+import {
+  client as lensClient,
+  refresh,
+  challenge,
+  authenticate,
+} from "../clientApi";
+import { ethers } from "ethers";
+
+const useLensToken = (setToken, setIsLensConnected) => {
+  // Local storage token management
+  const storeToken = useCallback(
+    (token, refresh) => {
+      setToken(token);
+      window.localStorage.setItem("lens-auth-token", token);
+      window.localStorage.setItem("lens-refresh-token", refresh);
+      setIsLensConnected(true);
+    },
+    [setToken, setIsLensConnected]
+  );
+  const removeToken = useCallback(() => {
+    setToken("");
+    window.localStorage.removeItem("lens-auth-token");
+    window.localStorage.removeItem("lens-refresh-token");
+    setIsLensConnected(false);
+  }, [setToken, setIsLensConnected]);
+
+  // Lens refresh functions
+  // Connect
+  const lensLogin = async (lensaddress) => {
+    try {
+      const challengeInfo = await lensClient.query({
+        query: challenge,
+        variables: { address: lensaddress },
+      });
+      const provider = new ethers.providers.Web3Provider(window.ethereum);
+      const signer = provider.getSigner();
+      const signature = await signer.signMessage(
+        challengeInfo.data.challenge.text
+      );
+      const authData = await lensClient.mutate({
+        mutation: authenticate,
+        variables: {
+          address: lensaddress,
+          signature,
+        },
+      });
+      const {
+        data: {
+          authenticate: { accessToken, refreshToken },
+        },
+      } = authData;
+      setToken(accessToken, refreshToken);
+    } catch (err) {
+      window.localStorage.removeItem("lens-auth-token");
+      console.log("Error signing in: ", err);
+    }
+  };
+  const refreshRequest = useCallback(async () => {
+    // Check for Lens Token already in local storage
+    const localStorage = window.localStorage;
+    const localToken = localStorage.getItem("lens-auth-token");
+    // If there's a token the refresh request is started
+    if (localToken) {
+      const localRefreshToken = localStorage.getItem("lens-refresh-token");
+      if (localRefreshToken) {
```

Comment:
What do you think about removing this nested if statment...
```suggestion
    const localToken = localStorage.getItem("lens-auth-token");
    const localRefreshToken = localStorage.getItem("lens-refresh-token");
    
    // If there's a token the refresh request is started
    if (localToken && localRefreshToken) {
```

Reply:
Good call!